app/controllers/solutions_controller.rb

```diff
@@ -17,7 +17,7 @@ def index
     @total_solutions = @solutions.count
     @solutions = @solutions.page(params[:page]).per(21)
     @reaction_counts = Reaction.where(solution_id: @solutions.map(&:id)).group(:solution_id, :emotion).count
-    @comment_counts = Reaction.where(solution_id: @solutions.map(&:id)).with_comments.group(:solution_id).count
+    @comment_counts = SolutionComment.where(solution_id: @solutions.map(&:id)).group(:solution_id).count
```

Comment:
Aren't those a nice use case for `counter_cache`s?

Reply:
Yeah, they probably are. I'm wary of adding more columns to solutions but this might be a good use case.